File: backend/src/test-mutations.ts
// Quick test script to verify mutation generation
import { MutationGenerator } from './services/mutationGenerator.js';

const generator = new MutationGenerator();
const testPrompt = 'You are a helpful assistant. Only answer questions about cooking.';

console.log('🧪 Testing Mutation Generator\n');
console.log('Original Prompt:');
console.log(`"${testPrompt}"\n`);

const mutations = generator.generateAll(testPrompt);

console.log('📊 Generated Mutations:\n');

Object.entries(mutations).forEach(([category, prompts]) => {
  console.log(`\n${category.toUpperCase()} (${prompts.length} mutations):`);
  console.log('─'.repeat(60));
  prompts.forEach((mutation: string, i: number) => {
    console.log(`\n${i + 1}. ${mutation.substring(0, 100)}${mutation.length > 100 ? '...' : ''}`);
  });
});

console.log('\n\n✅ Mutation generation working correctly!');
console.log(`Total mutations: ${Object.values(mutations).flat().length}`);

File: backend-python/test_mutations.py
"""Quick test script to verify mutation generation"""

from app.services.mutation_generator import MutationGenerator

generator = MutationGenerator()
test_prompt = "You are a helpful assistant. Only answer questions about cooking."

print("🧪 Testing Mutation Generator\n")
print("Original Prompt:")
print(f'"{test_prompt}"\n')

mutations = generator.generate_all(test_prompt)

print("📊 Generated Mutations:\n")

for category, prompts in mutations.items():
    print(f"\n{category.upper()} ({len(prompts)} mutations):")
    print("─" * 60)
    for i, mutation in enumerate(prompts, 1):
        preview = mutation[:100] + "..." if len(mutation) > 100 else mutation
        print(f"\n{i}. {preview}")

print("\n\n✅ Mutation generation working correctly!")
print(f"Total mutations: {sum(len(p) for p in mutations.values())}")
